src/base/Roles/CrossChain/CrossChainTellerBase.sol

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import {TellerWithMultiAssetSupport} from "../TellerWithMultiAssetSupport.sol";
+import "../../../interfaces/ICrossChainTeller.sol";
+
+/**
+ * @title CrossChainTellerBase
+ * @notice Base contract for the CrossChainTeller, includes functions to overload with specific bridge method
+ */
+abstract contract CrossChainTellerBase is ICrossChainTeller, TellerWithMultiAssetSupport{
+    
+    mapping(uint32 => Chain) public selectorToChains;
+
+    constructor(address _owner, address _vault, address _accountant, address _weth)
+        TellerWithMultiAssetSupport(_owner, _vault, _accountant, _weth)
+    {
+
+    }
+
+    // ========================================= ADMIN FUNCTIONS =========================================
+    /**
+     * @notice Add a chain to the teller.
+     * @dev Callable by OWNER_ROLE.
+     * @param chainSelector The chain selector to add.
+     * @param allowMessagesFrom Whether to allow messages from this chain.
+     * @param allowMessagesTo Whether to allow messages to this chain.
+     * @param targetTeller The address of the target teller on the other chain.
+     * @param messageGasLimit The gas limit for messages to this chain.
+     */
+    function addChain(
+        uint32 chainSelector,
+        bool allowMessagesFrom,
+        bool allowMessagesTo,
+        address targetTeller,
+        uint64 messageGasLimit
+    ) external requiresAuth {
+        if (allowMessagesTo && messageGasLimit == 0) {
+            revert CrossChainTellerBase_ZeroMessageGasLimit();
+        }
+        selectorToChains[chainSelector] = Chain(allowMessagesFrom, allowMessagesTo, targetTeller, messageGasLimit);
+
+        emit ChainAdded(chainSelector, allowMessagesFrom, allowMessagesTo, targetTeller, messageGasLimit);
+    }
+
+    /**
+     * @notice Remove a chain from the teller.
+     * @dev Callable by MULTISIG_ROLE.
```

Comment:
For the @dev comments mentioning roles, my thinking is that we will have a single owner role (the multisig) that is able to call privileged functions. 

Would be misleading to separate the `MULTSIG_ROLE` and `OWNER_ROLE` in the natspec comments.